topics/admin/tutorials/job-destinations/tutorial.md

```diff
@@ -425,33 +426,49 @@ If you don't want to write dynamic destinations yourself, Dynamic Tool Destinati
 >    > {: data-cmd="true"}
 >    {: .code-in}
 >
-{: .hands_on}
 
-## Testing the DTD
+# Basic access controls
 
-Our rule specified that any invocation of the `testing` tool with an input dataset with size <16 bytes would run on the 1 core destination, whereas any with >= 16 bytes would run on the 2 core destination.
+You may wish to apply some basic restrictions on which users are allowed to run specific tools. TPV accomodates user and role specific rules. In addition, TPV supports tagging of tools, users, roles and destinations. These tags
+can be matched up so that only desired combinations are compatible with each other. While these mechanisms are detailed in the TPV documentation, we will choose a different problem that highlights some other capabilities
+- restricting a tool so that only an admin can execute that tool.
 
-> <hands-on-title>Testing the DTD</hands-on-title>
+> <hands-on-title>Using conditionals to restrict a tool to admins only</hands-on-title>
 >
-> 1. Create a dataset using the upload paste tool with a few (<16) characters
+> 2. Edit your `files/galaxy/config/tpv_rules_local.yml` and add the following rule.
 >
-> 2. Create a dataset using the upload paste tool with >16 characters
->
-> 3. Run the `Testing Tool` on both datasets.
+>    {% raw %}
+>    ```diff
+>    --- a/files/galaxy/config/tpv_rules_local.yml
+>    +++ b/files/galaxy/config/tpv_rules_local.yml
+>    @@ -8,6 +8,14 @@ tools:
+>         mem: cores * 4
+>       testing:
+>         cores: 2
+>    +    rules:
+>    +      - id: admin_only_testing_tool
+>    +        if: |
+>    +          # Only allow the tool to be executed if the user is an admin
+>    +          admin_users = app.config.get( "admin_users", "" ).split( "," )
+>    +          # last line in block must evaluate to a value - which determines whether the TPV if conditional matches or not
+>    +          not user or user.email not in admin_users
+>    +        fail: Unauthorized. Only admins can execute this tool.
+>
+>     destinations:
+>       local_env:
+>    {% endraw %}
+>    ```
+>    {: data-commit="TPV admin only tool"}
 >
-{: .hands_on}
-
-> ```bash
-> 3.sh
-> ```
-> {: data-test="true"}
-{: .hidden}
```

Comment:
is there a reason to delete this test?

Reply:
Thanks for the catch. That should be restored.